### PR TITLE
Entities

### PR DESCRIPTION
**Pull request by cher-ning**
set up Collection and CollectionFactory interfaces, which are implemented through CommonRestriction, CommonInventory, and CommonRestrictionFactory, CommonInventoryFactory respectively. Set up CommonUser constructor to be used in CommonUserFactory, which will be implemented later on.

### CODE REVIEW

File: src/entity/CommonInventoryFactory.java [hunk 1]
```diff
@@ -1,4 +1,20 @@
...
+    public CommonInventory create(List<String> items) {
+        CommonInventory inventory = new CommonInventory();
+        List listIngredient = new ArrayList<CommonIngredient>();
+        for (int i = 0; i < items.size(); i++) {
+            Ingredient item = factory.create(items.get(i));
+            listIngredient.add(item);
+        }
+        inventory.addItems(listIngredient);
+        return inventory;
+    }
```
Comment (WujiMikeChen) [inline src/entity/CommonInventoryFactory.java:19]: for non trivial code please comment what the code is doing

**Review by WujiMikeChen — CHANGES_REQUESTED** [review]

File: src/entity/CommonInventory.java [hunk 2]
```diff
@@ -1,4 +1,26 @@
...
 
-public class CommonInventory {
+import java.util.ArrayList;
+import java.util.List;
+
+public class CommonInventory implements Collection {
+
+    private List<Ingredient> inventory;
+
+    CommonInventory() { this.inventory = new ArrayList(); }
```
Comment (JohnZHANGFZ) [inline src/entity/CommonInventory.java:10]: consider to make the constructor to be public

File: src/entity/CommonInventory.java [hunk 3]
```diff
@@ -1,4 +1,26 @@
...
+
+    @Override
+    public void addItems(List<Ingredient> ingredient) {
+        for (int i=0; i < ingredient.size(); i++) {
+            inventory.add(ingredient.get(i));
+        }
+    }
+
+    @Override
+    public void deleteItems(List<Ingredient> ingredient) {
```
Comment (JohnZHANGFZ) [inline src/entity/CommonInventory.java:23]: need to implement this method later

**Review by JohnZHANGFZ — CHANGES_REQUESTED** [review]

File: src/entity/CommonRecipe.java [hunk 4]
```diff
@@ -2,7 +2,7 @@
 
 import java.util.List;
 
-public class CommonRecipe implements Recipe{
+public class CommonRecipe implements Recipe {
     private final Integer rank;
```
Comment (JohnZHANGFZ) [inline src/entity/CommonRecipe.java:6]: This variable may not necessary if we get the result directly from API

File: src/entity/CommonRestriction.java [hunk 5]
```diff
@@ -20,10 +20,10 @@ public void addItems(List<Ingredient> ingredient) {
```
Comment (WujiMikeChen) [inline src/entity/CommonRestriction.java:20]: Given what input what does this chunk of code do, please comment

**Review by WujiMikeChen — APPROVED** [review]

**Pull request merged**